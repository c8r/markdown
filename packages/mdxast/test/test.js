const fs = require('fs')
const unified = require('unified')
const remark = require('remark-parse')
const rehype = require('remark-rehype')
const matter = require('remark-frontmatter')
const html = require('rehype-stringify')
const blocks = require('remark-parse/lib/block-elements.json')
const babel = require('@babel/core')

const toMdx = require('..')

const fixture = {
  basic: fs.readFileSync('test/fixtures/basic.md', 'utf8'),
  renderProps: fs.readFileSync('test/fixtures/render-props.md', 'utf8'),
}

const parseFixture = str => {
  const options = {
    blocks,
    matter: {
      type: 'yaml',
      marker: '-',
    },
  }

  const parser = unified()
    .use(remark, options)
    .use(matter, options.matter)
    .use(toMdx, options)
    .use(rehype)
    .use(html)

  return parser.processSync(str).contents
}

test('it parses a file', () => {
  const result = parseFixture(fixture.basic)
  expect(result).toMatchSnapshot()
})

test('using render props', () => {
  const code = parseFixture(fixture.renderProps).replace(/&#x3C;/g, '<')
  babel.parse(code, {
    plugins: ['@babel/plugin-syntax-jsx'],
  })
})
