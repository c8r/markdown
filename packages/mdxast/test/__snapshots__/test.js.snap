// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it parses a file 1`] = `
"import {
Foo,
Bar
} from './ui'
import Baz from './ui/baz'
import FooBar from './ui/foo-bar'
<h1>Hello, world!</h1>
<p>I'm an awesome paragraph.</p>
&#x3C;Foo bg='red'>
  &#x3C;Bar />
  {props.hello}
&#x3C;/Foo>
<h3>An inline image</h3>
<p><a href=\\"https://c8r-x0.s3.amazonaws.com/lab-components-macbook.jpg\\">https://c8r-x0.s3.amazonaws.com/lab-components-macbook.jpg</a></p>
<h3>Shopping list</h3>
<ul>
<li>Apples</li>
<li>Bananas</li>
<li>Oranges</li>
</ul>"
`;

exports[`using render props component 1`] = `
"import { Foo } from './Foo'
import { Bar } from './Bar'
<h1>Some test</h1>
&#x3C;Foo>
  {() => (
    &#x3C;div>helo&#x3C;/div>
  )}
&#x3C;/Foo>
&#x3C;Bar>
  &#x3C;Foo>
    {() => {
      const message = 'helo'
  return (
    &#x3C;div>{message}&#x3C;/div>
  )
}}
  &#x3C;/Foo>
&#x3C;/Bar>
&#x3C;Bar>
  &#x3C;Foo>
    {() => {
      const message = 'helo'
  return (
    &#x3C;Foo>
      {() => {
        const name = 'john'

        return (
          &#x3C;div>{message} {john}&#x3C;/div>
        )
      }}
    &#x3C;/Foo>
  )
}}
  &#x3C;/Foo>
&#x3C;/Bar>"
`;
